serde-support/src/main/java/io/micronaut/serde/support/serdes/EnumSerde.java

```diff
@@ -145,12 +142,16 @@ public EnumCreatorDeserializer(
     @NonNull
     private E transform(Object v) {
         try {
-            return (E) deserializableIntrospection.instantiate(v);
+            Object[] args = new Object[1];
+            args[0] = v;
+            return (E) deserializableIntrospection.instantiate(!allowNull, args);
         } catch (IllegalArgumentException e) {
             if (v instanceof String) {
                 String string = (String) v;
                 try {
-                    return (E) deserializableIntrospection.instantiate(string.toUpperCase(Locale.ENGLISH));
+                    Object[] args = new Object[1];
+                    args[0] = string.toUpperCase(Locale.ENGLISH);
```

Comment:
initialise on the same line `new Object[] { val }`

Reply:
Done via https://github.com/micronaut-projects/micronaut-serialization/pull/492